src/Compilers/Core/MSBuildTask/AssemblyResolution.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Roslyn.Utilities;
+using System;
+using System.IO;
+using System.Reflection;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    // TODO: Remove once https://github.com/Microsoft/msbuild/issues/1309 is fixed.
+    internal static class AssemblyResolution
+    {
+        static AssemblyResolution()
+        {
+            CorLightup.Desktop.AddAssemblyResolveHandler(ResolveAssembly);
+        }
+
+        public static void Install()
+        {
+            // empty, just to trigger static ctor
+        }
+
+        internal static Assembly ResolveAssembly(string assemblyDisplayName, Assembly requestingAssemblyOpt)
+        {
+            var name = new AssemblyName(assemblyDisplayName);
+            return TryRedirect(name) ? Assembly.Load(name) : null;
+        }
+
+        private static readonly byte[] s_b03f5f7f11d50a3a = new byte[] { 0xb0, 0x3f, 0x5f, 0x7f, 0x11, 0xd5, 0x0a, 0x3a };
+        private static readonly byte[] s_b77a5c561934e089 = new byte[] { 0xb7, 0x7a, 0x5c, 0x56, 0x19, 0x34, 0xe0, 0x89 };
+
+        private static bool TryRedirect(AssemblyName name)
+        {
+            switch (name.Name)
```

Comment:
Have you verified that all of these items are still necessary?  I recently made considerable changes to this DLL and removed dependencies like System.Collections.Immutable.  Yet it's still listed below.